File: TODO list/frontend/src/types/todo.ts
export interface Todo {
  id?: number;
  title: string;
  description?: string;
  completed: boolean;
  category: string;
  priority: number;
  dueDate?: string;
  createdAt?: string;
  updatedAt?: string;
}

export interface TodoForm {
  title: string;
  description: string;
  category: string;
  priority: number;
  dueDate: string;
}

File: TODO list/todolist/src/main/java/com/example/todolist/model/Todo.java
package com.example.todolist.model;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Data;
import java.time.LocalDateTime;

@Data
public class Todo {
    private Long id;
    private Long userId;
    private String title;
    private String description;
    private Boolean completed;
    private String category;
    private Integer priority;
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime dueDate;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}